src/zenml/pipelines/base_pipeline.py

```diff
@@ -260,7 +269,22 @@ def requirements(self) -> Set[str]:
                     f"'{integration_name}'."
                 ) from e
 
-        if self.requirements_file and fileio.exists(self.requirements_file):
+        if (
+            isinstance(self._requirements, str)
+            and self._requirements
```

Comment:
Why does their need to be a `self._requirements` here

Reply:
to check it's not `None`

Reply:
I think that should already be covered by the `isinstance` check

Reply:
If string1 is "", `isinstance(string1, str)` evaluates to `True`, but `bool("")` evaluates to `False`. Should I not be trying to catch that case? I think I'm misunderstanding something here.